### PR TITLE
new implementation of getRandomSample (sample2ddf) to make it work with complex data type in DDF

### PR DESCRIPTION
**Pull request by namma**

### CODE REVIEW

**Comment by khangich** [comment]
ok to test

**Comment by namma** [comment]
@mbbui @Huandao0812 please review and test

**Comment by mbbui** [comment]
retest this please

**Comment by namma** [comment]
retest this please

File: spark/src/main/scala/io/ddf/spark/util/SparkUtils.scala [hunk 1]
```diff
@@ -123,25 +123,86 @@ object SparkUtils {
...
+//  }
 
   /**
    *
    * @param df
+   * @param sep the separator to separate adjacent column
    * @return an Array of string showing the dataframe with complex column-object replaced by json string
    */
-  def jsonForComplexType(df: DataFrame, sep: String): Array[String] = {
+  def df2txt(df: DataFrame, sep: String): Array[String] = {
```
Comment (mbbui) [inline spark/src/main/scala/io/ddf/spark/util/SparkUtils.scala]: let change this to sparkdf2txt, so it is clear.

**Pull request merged**